base_export_email/ir_actions.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Authors: Laetitia Gangloff
+#    Copyright (c) 2015 Acsone SA/NV (http://www.acsone.eu)
+#    All Rights Reserved
+#
+#    WARNING: This program as such is intended to be used by professional
+#    programmers who take the whole responsibility of assessing all potential
+#    consequences resulting from its eventual inadequacies and bugs.
+#    End users who are looking for a ready-to-use solution with commercial
+#    guarantees and support are strongly advised to contact a Free Software
+#    Service Company.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+from openerp.addons.web.controllers import main
+from openerp.tools.safe_eval import safe_eval
+import time
+import base64
+
+
+class actions_server(orm.Model):
+    _inherit = "ir.actions.server"
+
+    def __init__(self, pool, cr):
+        super(actions_server, self).__init__(pool, cr)
+        # add state 'export_email'
+        self._columns['state'].selection.append(('export_email',
+                                                 _('Export data by email')))
+        return
+
+    _columns = {'model': fields.related('model_id', 'model',
+                                        type="char", size=256, string='Model'),
+                'filter_id': fields.many2one(
+                    'ir.filters', string='Filter', ondelete='restrict'),
+                'email_template_id': fields.many2one(
+                    'email.template', string='Template', ondelete='restrict'),
+                'saved_export_id': fields.many2one(
+                    'ir.exports', string='Saved export', ondelete='restrict'),
+                'fields_to_export': fields.text(
+                    'Fields to export',
+                    help="The list of fields to export. \
+                          The format is ['name', 'seller_ids/product_name']")
+                }
```

Comment:
The relationship between Saved export and Fields to export may not be obvious for the user.